sdk/dapp-kit/src/components/WalletProvider.tsx

```diff
@@ -93,6 +95,28 @@ export function WalletProvider({
 
 	useUnsafeBurnerWallet(enableUnsafeBurner);
 
+	useEffect(() => {
+		if (!autoConnect || walletState.currentWallet) {
+			return;
+		}
+
+		(async function autoConnectWallet() {
+			const connectionInfo = await getMostRecentWalletConnectionInfo(storageAdapter, storageKey);
+			const { walletName, accountAddress } = connectionInfo || {};
+			const wallet = walletName
+				? walletState.wallets.find((wallet) => wallet.name === walletName)
+				: null;
+
+			if (wallet) {
+				connectWallet(dispatch, storageAdapter, storageKey, {
```

Comment:
Why not use the `useConnect` hook? My guess is that it's because we're actually defining the context here, so you can't use it because it consumes the context. Fair but also, what if you went the route of making an `<Autoconnect />` component (instead of the `useAutoconnect` hook I reference above), and conditionally render it based on the `autoConnect` prop, and just have it under the context provider and able to use all the normal machinery? I think this also would let you avoid some of the abstraction you do here like separating this out in general.

Reply:
> fair but also, what if you went the route of making an <Autoconnect /> component (instead of the useAutoconnect hook I reference above), and conditionally render it based on the autoConnect prop, and just have it under the context provider and able to use all the normal machinery?

I actually did exactly this the first time around, but I ran into more dependency cycle issues that made me rage quit and go with this approach 😆. Cycle example:
- `useWalletContext` relies on `WalletProvider`
- `useConnectWallet` relies on `useWalletContext`
- `WalletProvider` can't import `useConnectWallet` or any component using that hook

I could make a `WalletProviderContainer` type component that renders the WalletProvider with the Autoconnect component as a child, or I could just hoist it all up to the future `DAppKitProvider` (this was the second solution I mentioned in the PR description). What do you think?

Reply:
If you make an Autoconnect component and chuck it into a separate file it seems like it solves these problems without issue? You can even render it in the provider here in A Fragment

Reply:
That also results in a dependency cycle 🤔. I think I figured it out though after making a shoddy diagram haha, `WalletContext` being located in `WalletProvider` was the source of my dependency cycle issues:

Cycle:
<img width="828" alt="image" src="https://github.com/MystenLabs/sui/assets/7453188/1520ef5d-dc62-4395-a6a3-e22cb9107c4f">

Fixt:
<img width="960" alt="image" src="https://github.com/MystenLabs/sui/assets/7453188/a67ce76f-0141-492c-b880-17b648aa06d8">

Reply:
Ah yeah separating out the context is probably a good idea to avoid these kinds of problems from popping up in general.